extensions/main/deployment/src/main/java/org/apache/camel/quarkus/main/deployment/SubstrateProcessor.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.main.deployment;
+
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.substrate.ReflectiveClassBuildItem;
+import org.apache.camel.quarkus.core.deployment.CamelSupport;
+import org.jboss.jandex.IndexView;
+
+public class SubstrateProcessor {
```

Comment:
Why do we need two Processors? It would be nice to figure out from JavaDoc which tasks belong where.

Reply:
what do you mean by two processors ?

Reply:
There is `org.apache.camel.quarkus.main.deployment.BuildProcessor` and `org.apache.camel.quarkus.main.deployment.SubstrateProcessor`. Both contain build steps and it is not clear to me what would be the criteria to decide for the current and future build steps to decide into which one they belong?

Reply:
`SubstrateProcessor` is meant to contains build steps related to substrate like reflection, substrate properties and so on. The distinction is juts logical so one know where to look depending of the context.

Reply:
Thanks, makes sense. How about adding a class level JavaDoc something like `Produces BuildItems from the io.quarkus.deployment.builditem.substrate package` ?